x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/stats/StatsPersister.java

```diff
@@ -46,13 +46,13 @@ public void persistWithRetry(ToXContentObject result, Function<String, String> d
                 docIdSupplier.apply(jobId),
                 true,
                 () -> true,
-                errorMsg -> auditor.error(jobId,
-                    "failed to persist result with id [" + docIdSupplier.apply(jobId) + "]; " + errorMsg)
+                retryMessage -> LOGGER.debug("[{}] failed to persist result with id [{}]; {}", jobId, docIdSupplier.apply(jobId), retryMessage)
             );
         } catch (IOException ioe) {
             LOGGER.error(() -> new ParameterizedMessage("[{}] Failed serializing stats result", jobId), ioe);
         } catch (Exception e) {
             LOGGER.error(() -> new ParameterizedMessage("[{}] Failed indexing stats result", jobId), e);
+            auditor.error(jobId, "Failed indexing stats result with id [" + docIdSupplier.apply(jobId) + "]; " + e.getMessage());
```

Comment:
Double-checking: should it be semicolon or colon in this string: `"]; "`?

Reply:
Good question. I tend avoid semi-colons as I don't know how to use them correctly. 

This one was copied from the old usage above

Reply:
@dimitris-athanasiou you wrote this code originally can you comment on `;` vs `:` please